napari_aicssegmentation/controller/batch_processing_controller.py

```diff
@@ -13,8 +17,53 @@ def __init__(self, application: IApplication, workflow_engine: WorkflowEngine):
         self._workflow_engine = workflow_engine
         self._view = BatchProcessingView(self)
 
+        # Should these go into a model?
+        self.input_folder: Path = None
+        self.output_folder = None
+        self.selected_index = None
+        self.workflow_config = None
+
     def index(self):
         self.load_view(self._view)
 
     def run_batch(self):
-        return super().run_batch()
+        # workflow = self.get_batch_workflow()
+        # workflow.process_all()
+        self.worker = create_worker(self.run_batch_async)
+        self.worker.start()
+
+    def run_batch_async(self):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            workflow = self.get_batch_workflow()
+            workflow.process_all()
+
+    def ready_to_process(self):
+        if not self.workflow_config:
+            return False
+        elif not self.input_folder:
+            return False
+        elif not self.output_folder:
+            return False
+        else:
+            return True
+
+    def select_config(self, selected_config):
```

Comment:
Reminder to add controller public methods to the interface when you're fully happy with the design.